### PR TITLE
Add GHC-9.8.1-alpha2

### PR DESCRIPTION
**Pull request by bgamari**

### CODE REVIEW

File: ghcup-prereleases-0.0.7.yaml [hunk 1]
```diff
@@ -952,6 +952,121 @@ ghcupDownloads:
...
+              dlSubdir: ghc-9.8.0.20230809-aarch64-apple-darwin
+              dlUri: https://downloads.haskell.org/~ghc/9.8.0.20230809/ghc-9.8.0.20230809-aarch64-apple-darwin.tar.xz
+          Linux_UnknownLinux:
+            unknown_versioning:
+              dlHash: fed0c4372f0288d62e41b4af379a8c4d18ad2f9792973246f3650395911361e9
+              dlOutput: ghc-9.8.0.20230809-aarch64-linux-deb10.tar.xz
+              dlSubdir: ghc-9.8.0.20230809-aarch64-unknown-linux
+              dlUri: https://downloads.haskell.org/~ghc/9.8.0.20230809/ghc-9.8.0.20230809-aarch64-deb10-linux.tar.xz
+      viChangeLog: https://downloads.haskell.org/~ghc/9.8.0.20230809/docs/users_guide/9.8.1-notes.html
+      viReleaseDay: '2023-03-09'
```
Comment (andreasabel) [inline ghcup-prereleases-0.0.7.yaml:1056]: Is there a typo in the date?

**Comment by phadej** [comment]
The alpha2 also doesn't show in the list when I do `ghcup list`. (`master` is not updated?!)

**Comment by andreasabel** [comment]
I had to manually do a `wget` of the prerelease file into `.ghcup/cache`, because `ghcup` would not update it automatically, for reasons unknown to me.

**Comment by phadej** [comment]
> I had to manually do a `wget` of the prerelease file into `.ghcup/cache`, because `ghcup` would not update it automatically, for reasons unknown to me.

```
% ~/.ghcup/bin/ghcup list
[ Info  ] downloading: https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-0.0.7.yaml as file /home/phadej/.ghcup/cache/ghcup-0.0.7.yaml
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[ Info  ] downloading: https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-prereleases-0.0.7.yaml as file /home/phadej/.ghcup/cache/ghcup-prereleases-0.0.7.yaml
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
```

my ghcup uses `master` branch, because I added `master` branch. (That's URL which `ghcup` [docs](https://www.haskell.org/ghcup/guide/) mention).

This PR was into `develop` branch, and I don't know what's the process of updating `master`.

**Comment by bgamari** [comment]
Thanks for the ping, @phadej. I have now merged `develop` into `master`.

 I need to better document the process for updating `ghcup` to ensure this doesn't happen again in the future.

File: ghcup-prereleases-0.0.7.yaml [hunk 2]
```diff
@@ -952,6 +952,121 @@ ghcupDownloads:
...
+      viSourceDL:
+        dlHash: ecef10e9796082eaf3f18ece67fe5fcfedfd0baa77d34c1f772e37aabda3fa1c
+        dlOutput: ghc-9.8.0.20230809-src.tar.xz
+        dlSubdir: ghc-9.8.0.20230809
+        dlUri: https://downloads.haskell.org/~ghc/9.8.0.20230809/ghc-9.8.0.20230809-src.tar.xz
+      viTags:
+      - Prerelease
```
Comment (hasufell) [inline ghcup-prereleases-0.0.7.yaml:1063]: @bgamari this is wrong too. It needs to be `LatestPrerelease` and the old one changed to `Prerelease`.

**Pull request merged**